templates/tower-project.yaml

```diff
@@ -0,0 +1,338 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >-
+  Stack for provisioning the AWS resources needed for
+  configuring compute environments in Nextflow Tower.
+
+Parameters:
+  ReadWriteAccessArns:
+    Type: List<String>
+    Default: ""
+    Description: >-
+      (Optional) List of IAM users and roles (strings) who
+      should have read- and write-access to the S3 bucket.
+      You can provide role ARNs or assumed-role ARNs.
+
+  ReadOnlyAccessArns:
+    Type: List<String>
+    Default: ""
+    Description: >-
+      (Optional) List of IAM users and roles (strings)
+      who should only have read-access to the S3 bucket.
+      You can provide role ARNs or assumed-role ARNs.
+
+  EnableDataLifeCycle:
+    Type: String
+    Description: Enabled to enable bucket lifecycle rule, default is Disabled
+    AllowedValues:
+      - Enabled
+      - Disabled
+    Default: Disabled
+
+  LifecycleDataTransition:
+    Type: Number
+    Description: >-
+      Number of days until S3 objects are moved to LifecycleDataStorageClass
+    Default: 30
+
+  LifecycleDataStorageClass:
+    Type: String
+    Description: S3 bucket objects will transition into this storage class
+    AllowedValues:
+      - DEEP_ARCHIVE
+      - INTELLIGENT_TIERING
+      - STANDARD_IA
+      - ONEZONE_IA
+      - GLACIER
+    Default: GLACIER
+
+  LifecycleDataExpiration:
+    Type: Number
+    Description: >-
+      Number of days (from creation) when objects are
+      deleted from S3 and the LifecycleDataStorageClass
+    Default: 365000
+
+  AdminArns:
+    Type: List<String>
+    Description: >-
+      List of IAM user and role ARNs (strings) who should have access to 
+      project-specific resources (e.g., private S3 bucket, KMS key, secrets). 
+  
+  NonAdminArns:
+    Type: List<String>
+    Description: >-
+      List of IAM user and role ARNs (strings) who shouldn't have access to 
+      project-specific resources (e.g., private S3 bucket, KMS key, secrets). 
```

Comment:
I confirmed that I don't have to explicitly deny the non-admin ARNs on the resource policy for the secret.